### ISSUE TITLE
Point not reading Z

### ISSUE THREAD

**Issue by danny3301** [rgeo/activerecord-postgis-adapter#245]
Using RGeo 3.1.4
Rails 4.2
ActiveRecord 4.2.4

I am currently having an issue with my geometry column:
t.geometry "point",              limit: {:srid=>4326, :type=>"point", :has_z=>true}

When I create a new point it creates the point just fine. When I update X or Y, the point updates just fine. It is when I just update Z that the point is not updated. It seems like it is not reading in the z-coordinate at all. 

Here is the column definition:
POINT --> --- !ruby/object:RGeo::Cartesian::PointImpl
factory: !ruby/object:RGeo::Cartesian::Factory
  has_z_coordinate: true
  has_m_coordinate: false
  srid: 4326
  lenient_assertions: false
  buffer_resolution: 1
  wkt_generator:
    tag_format: wkt11
    emit_ewkt_srid: 
    square_brackets: false
    convert_case: upper
  wkb_generator:
    type_format: wkb11
    emit_ewkb_srid: 
    hex_format: false
    little_endian: false
  wkt_parser:
    support_ewkt: false
    support_wkt12: false
    strict_wkt11: false
    ignore_extra_tokens: false
    default_srid: 
  wkb_parser:
    support_ewkb: false
    support_wkb12: false
    ignore_extra_bytes: false
    default_srid: 
wkt: Point Z (-119.546875 36.319049 900.0)

Am I missing something on this?

**Comment by PelagicDev** [comment]
Can you post your schema for the relevant table here?

**Comment by danny3301** [comment]
create_table "sites", force: :cascade do |t|
    t.string   "name",               limit: 255,                                                         null: false
    t.date     "activated_at"
    t.date     "deactivated_at"
    t.text     "comments"
    t.float    "latitude",                                                                               null: false
    t.float    "longitude",                                                                              null: false
    t.float    "elevation",                                                                              null: false
    t.string   "elevation_type",     limit: 255
    t.string   "location_source",    limit: 255
    t.geometry "point",              limit: {:srid=>4326, :type=>"point", :has_z=>true},                 null: false
  end

**Comment by PelagicDev** [comment]
Your issue looks to be on the last line: `t.geometry "point", limit: {:srid=>4326, :type=>"point", :has_z=>true}, null: false`

try doing a migration that removes that column and adds a new column using `st_point` as the column type instead of `point`.

**Comment by PelagicDev** [comment]
your migration might look something like this, although I can't say if this is 100% correct for your app:
```
def change
    remove_column :sites, :point, :point
    add_column :sites, :point, :st_point, geographic: true, null: false
  end
```

**Comment by teeparham** [comment]
This may be fixed by #280 / #281.
